packages/tpetra/core/test/FECrsMatrix/FECrsMatrix_UnitTests.cpp

```diff
@@ -292,13 +302,92 @@ TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL( FECrsMatrix, Assemble1D, LO, GO, Scalar, Node
   TPETRA_GLOBAL_SUCCESS_CHECK(out,comm,success)
 }
 
+////
+TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL( FECrsMatrix, Assemble1D_Kokkos, LO, GO, Scalar, Node )
+{
+  using exec_space = typename Node::execution_space;
+  using range_type = Kokkos::RangePolicy<exec_space, LO>;
+  using FEMAT = typename Tpetra::FECrsMatrix<Scalar,LO,GO,Node>;
+  using CMAT = typename Tpetra::CrsMatrix<Scalar,LO,GO,Node>;
+  using FEG = typename Tpetra::FECrsGraph<LO,GO,Node>;
+  using ImplScalarType = typename FEMAT::impl_scalar_type;
+
+  // get a comm
+  RCP<const Comm<int> > comm = getDefaultComm();
+  
+  // Generate a mesh
+  size_t numLocal = 10;
+  GraphPack<LO,GO,Node> pack;
+  generate_fem1d_graph(numLocal,comm,pack);
+
+  // Make the graph    
+  // FIXME: We should be able to get away with 3 for StaticProfile here, but we need 4 since duplicates are
+  // not being handled correctly. 
+  RCP<FEG> graph = rcp(new FEG(pack.uniqueMap,pack.overlapMap,4));
+
+  graph->beginFill();
+  for(size_t i=0; i<(size_t)pack.element2node.size(); i++) {
```

Comment:
Prefer C++ - style casts.  Also, you use `pack.element2node.size()` and `pack.element2node[i].size()` a lot; consider naming them.
```c++
// use () instead of = to avoid signed/unsigned warnings.
const size_t outer_size (pack.element2node.size());
for (size_t i = 0; i < outer_size; ++i) {
  const size_t inner_size (pack.element2node[i].size());
  for (size_t j = 0; j < inner_size; ++j) {
    // ... etc. ...
  }
}
```